uniqueidentifier_test.go

```diff
@@ -9,56 +9,76 @@ import (
 	"testing"
 )
 
-func TestUniqueIdentifier(t *testing.T) {
+func TestUniqueIdentifierScanBytes(t *testing.T) {
+	t.Parallel()
 	dbUUID := UniqueIdentifier{0x67, 0x45, 0x23, 0x01,
 		0xAB, 0x89,
 		0xEF, 0xCD,
 		0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF,
 	}
+	uuid := UniqueIdentifier{0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF, 0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF}
 
+	var sut UniqueIdentifier
+	scanErr := sut.Scan(dbUUID[:])
+	if scanErr != nil {
+		t.Fatal(scanErr)
+	}
+	if sut != uuid {
+		t.Errorf("bytes not swapped correctly: got %q; want %q", sut, uuid)
+	}
+}
+
+func TestUniqueIdentifierScanString(t *testing.T) {
+	t.Parallel()
 	uuid := UniqueIdentifier{0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF, 0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF}
 
-	t.Run("Scan", func(t *testing.T) {
-		t.Run("[]byte", func(t *testing.T) {
-			var sut UniqueIdentifier
-			if err := sut.Scan(dbUUID[:]); err != nil {
-				t.Fatal(err)
-			}
-			if sut != uuid {
-				t.Errorf("bytes not swapped correctly: got %q; want %q", sut, uuid)
-			}
-		})
-
-		t.Run("string", func(t *testing.T) {
-			var sut UniqueIdentifier
-			if err := sut.Scan(uuid.String()); err != nil {
-				t.Fatal(err)
-			}
-			if sut != uuid {
-				t.Errorf("string not scanned correctly: got %q; want %q", sut, uuid)
-			}
-		})
-	})
-
-	t.Run("Value", func(t *testing.T) {
-		sut := uuid
-		v, err := sut.Value()
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		b, ok := v.([]byte)
-		if !ok {
-			t.Fatalf("(%T) is not []byte", v)
-		}
-
-		if !bytes.Equal(b, dbUUID[:]) {
-			t.Errorf("got %q; want %q", b, dbUUID)
-		}
-	})
+	var sut UniqueIdentifier
+	scanErr := sut.Scan(uuid.String())
+	if scanErr != nil {
+		t.Fatal(scanErr)
+	}
+	if sut != uuid {
+		t.Errorf("string not scanned correctly: got %q; want %q", sut, uuid)
+	}
+}
+
+func TestUniqueIdentifierScanUnexpectedType(t *testing.T) {
+	t.Parallel()
+	var sut UniqueIdentifier
+	scanErr := sut.Scan(int(1))
+	if scanErr == nil {
+		t.Fatal(scanErr)
```

Comment:
>(scanErr) [](http://example.com/codeflow?start=9&length=9)

an error message would be more helpful than nil

Reply:
7ae8aa04d1018ae19f238fa383f91f0df2d89d96

Reply:
?